[18.0][FIX] delivery_price_method: Skip base functionality

david-s73 opened the pull request:
- The change has been made because it skipped the base functionality to check if the shipment has to be free or not if it exceeded the price set in the field `free_over` I leave the exact line of the Odoo code, as it is forcing the `delivery_type` to be the same as this in the `price_method` then does not make the `if` when it should be done.
[Odoo code](https://github.com/odoo/odoo/blob/18.0/addons/delivery/models/delivery_carrier.py#L282)

david-s73 committed:
[FIX] delivery_price_method: Skip base functionality

david-s73 commented:
@rousseldenis Can you check the fix when you can?

david-s73 commented:
@pedrobaeza Can you check the fix when you can?

rousseldenis commented on delivery_price_method/models/delivery_carrier.py:
```diff
@@ -21,7 +21,9 @@ def rate_shipment(self, order):
         previous_method = False
-        if self.price_method in ("fixed", "base_on_rule"):
+        if self.price_method in ("fixed", "base_on_rule") and (
+            not self.free_over or not self.amount or order.amount_total < self.amount
```
Why not using `self._compute_currency()` as in base code ?

david-s73 replied:
I understand that you mean changing the `order.amount` to `self._compute_currency()` right?

rousseldenis replied:
@david-s73 Yes, because `self.amount` should be compared with currency comparison.

david-s73 replied:
@rousseldenis I have already made the changes you requested.

david-s73 committed:
[FIX] delivery_price_method: 'Rule-based' problem

If the pricing method is 'rule-based' it does not take into account the ‘free_over’ field so if it receives a price higher than any rule it will get a base error message.

pedrobaeza commented:
Can you add a test for the case?

david-s73 commented:
@pedrobaeza I have already taken the tests.

pedrobaeza commented:
Why the extra commit?

david-s73 commented:
@pedrobaeza Because it's another fix but from the same module, I put it in this pr.

pedrobaeza commented:
Well, starting with, the commit message is not correct: https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#commit-message, and if it's other module, it's better to put it in another PR.

david-s73 commented:
@pedrobaeza I have already modified the commit message, and since it is from the same module, I have put it in the same pr.

david-s73 commented:
@pedrobaeza Merged?

pedrobaeza commented:
Pending from @rousseldenis' approval

Reyes4711-S73 reviewed (approved):
LGTM

Reyes4711-S73 commented:
@rousseldenis @OCA/logistics-maintainers Please, can you merge this PR?

david-s73 commented:
@rousseldenis Can you approve it?

rousseldenis commented:
@david-s73 sorry, I answered late but was very busy. One point is still questionning for me.

rousseldenis commented:
/ocabot merge minor
